src/mds/Server.cc

```diff
@@ -5722,6 +5722,32 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::return_bytes(string& value_str)
+{
+  if (value_str.length() < 1) {
+    return -CEPHFS_EINVAL;
+  }
+  unordered_map<char, int64_t> size_value = {{'k', 1L << 10}, {'m', 1L << 20}, 
+                                             {'g', 1L << 30}, {'t', 1L << 40}};
+  char last_char = tolower(value_str.back());
+  string value = value_str.substr(0, value_str.length() - 1);
+  if (size_value.find(last_char) != size_value.end()) {
+    try {
+      return (boost::lexical_cast<int64_t>(value)) * size_value[last_char];
+    } catch (boost::bad_lexical_cast const&) {
+      dout(10) << "invalid numberical value encountered: " << value << dendl;
+      return -CEPHFS_EINVAL;
+    }
+  } else {
+    try {
+      return boost::lexical_cast<int64_t>(value_str);
+    } catch (boost::bad_lexical_cast const&) {
+      dout(10) << "could not identify the data size: " << value_str << dendl;
```

Comment:
Isn't this error same as above, i.e., invalid numerical value ? If so, this error string sounds little confusing to me.

Reply:
it this is reached then it means the it is a single character(alphabetical) value or the data type specified is not supported e.g. `100P` while the above error takes care of something like `1f00m` or `f100k` like that. So they both handle different cases. If it's kind of confusing then I can improve the statements or add some comments in the code itself. What do you think?

Reply:
ok, but this also handles the case of `1f001` input not ending with a character, so I got confused. So the error message looks fine.